pkgs/by-name/ap/apt/package.nix

```diff
@@ -83,6 +85,10 @@ stdenv.mkDerivation (finalAttrs: {
     # the following error:
     # E: Unable to determine a suitable packaging system type
     ln -s ${lib.getBin dpkg}/bin/dpkg $out/bin/dpkg
+
+    # apt-key is a shell script, as such, we need to specify its dependencies,
+    # since $PATH could be controlled by other packages
+    patsh -f $out/bin/apt-key -s ${builtins.storeDir}
```

Comment:
I highly recommend putting as a parameter `nixStoreDir ? config.nix.storeDir or builtins.storeDir` (grep for storeDir to see how we do it in other packages or top-level).

Reply:
But why not just patchShebangs?

Reply:
apt-key depends a lot of external packages (sed/awk/gpg...etc) in addition to shebang fixes. I tried to `makeWrapper` but the required `PATH` grows very fast and I'm still getting incorrect result which I was not able to figure out what was missing. I googled around to found `patsh` but none of the existing cases use parameterized `nixStoreDir` so I was not aware of it.

Reply:
I want to clarify that `patchShebangs` does fix `apt-key` when called from an interactive session. The issue I was getting is when `mmdebstrap` calls `apt-get`, which in turn calls `apt-key`. In that case the `PATH` is the one I `makeWrapper` specifically for `mmdebstrap`, and caused `apt-key` to fail.